File: 30-Days-of-Code/Day-6-Lets-Review/R/solution.r
lines <- readLines(file("stdin"))

print_func <- function(string){
  splitted <- unlist(strsplit(string, ""))
  firstHalf <- paste0(splitted[seq(1,length(splitted),2)], collapse = "")
  secondHalf <- paste0(splitted[seq(2,length(splitted),2)], collapse = "")
  c(firstHalf, secondHalf)
}

for (k in 2:length(lines)){
  cat(print_func(lines[[k]]), "\n")
}

File: 30-Days-of-Code/Day-6-Lets-Review/Rust/solution.rs
use std::io;
use std::io::prelude::*;

fn main() {
    let stdin = io::stdin();
    
    stdin.lock().lines().next();
    
    for line in stdin.lock().lines() {
        let s = line.unwrap();
        let mut firstHalf = String::new();
        let mut secondHalf = String::new();
        for (i, c) in s.char_indices() {
            if i % 2 == 0 {
                firstHalf.push(c);
            } else {
                secondHalf.push(c);
            }
        }
        println!("{} {}", firstHalf, secondHalf);
    }
}
